ingest/src/main/java/com/microsoft/azure/kusto/ingest/Utils.java

```diff
@@ -0,0 +1,110 @@
+package com.microsoft.azure.kusto.ingest;
+
+import com.microsoft.azure.kusto.ingest.exceptions.IngestionClientException;
+import com.microsoft.azure.storage.StorageException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.File;
+import java.io.IOException;
+import java.io.Writer;
+import java.lang.invoke.MethodHandles;
+import java.net.URISyntaxException;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+
+public class Utils {
+
+    private static final int COMPRESSED_FILE_MULTIPLIER = 11;
+    private final static Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    public static long estimateBlobRawSize(AzureStorageClient azureStorageClient, String blobPath) throws StorageException, URISyntaxException {
+        long blobSize = azureStorageClient.getBlobSize(blobPath);
+
+        return azureStorageClient.isCompressed(blobPath) ?
+                blobSize * COMPRESSED_FILE_MULTIPLIER : blobSize;
+    }
+
+    public static long estimateFileRawSize(AzureStorageClient azureStorageClient, String filePath) {
+        File file = new File(filePath);
+        long fileSize = file.length();
+
+        return azureStorageClient.isCompressed(filePath) ?
+                fileSize * COMPRESSED_FILE_MULTIPLIER : fileSize;
+    }
+
+    public static long resultSetToCsv(ResultSet resultSet, Writer writer, boolean includeHeaderAsFirstRow)
+            throws IngestionClientException {
+        final String LINE_SEPARATOR = System.getProperty("line.separator");
+
+        try {
+            String columnSeparator = "";
+
+            ResultSetMetaData metaData = resultSet.getMetaData();
+            int numberOfColumns = metaData.getColumnCount();
+
+            if (includeHeaderAsFirstRow) {
```

Comment:
code repetition from writeResultSetRow (not relevent if u listen to my above comment)